os-client/src/main/kotlin/io/provenance/os/client/OsClient.kt

```diff
@@ -211,28 +248,16 @@ open class OsClient(uri: URI) {
             throw responseObserver.error!!
         }
 
-        return responseObserver.get().toDomain()
+        return responseObserver.get()
     }
 
-    fun createPublicKey(publicKey: PublicKey): io.provenance.os.domain.PublicKey =
-        publicKeyBlockingClient.create(
-            PublicKeys.PublicKeyRequest.newBuilder()
-                .setPublicKey(ECUtils.convertPublicKeyToBytes(publicKey).toByteString())
-                .build()
-        ).toDomain()
-
-    fun deletePublicKey(publicKey: PublicKey) {
-        publicKeyBlockingClient.delete(
+    fun createPublicKey(publicKey: PublicKey): PublicKeys.PublicKeyResponse? =
+        publicKeyBlockingClient.add(
             PublicKeys.PublicKeyRequest.newBuilder()
-                .setPublicKey(ECUtils.convertPublicKeyToBytes(publicKey).toByteString())
+                .setPublicKey(publicKey.toPublicKeyProtoOS())
+                .setUrl("http://localhost") // todo: what is this supposed to be?
```

Comment:
does this need to be set at all?

Reply:
just wondering if that can become a future bug when we release to prod/test, cause there is no such thing as a localhost

Reply:
I was getting an error from OS if this was unset... not sure if that has changed now, if it has I vote we leave it out